### PR TITLE
flyte -> python

### PR DESCRIPTION
**Pull request by kennyworkman**

### CODE REVIEW

File: latch/cli/main.py [hunk 1]
```diff
@@ -181,6 +184,33 @@ def execute(params_file: Path, version: Union[str, None] = None):
...
+)
+def get_params(wf_name: Union[str, None], version: Union[str, None] = None):
+    """Generate a python parameter map for a workflow.
+
+    Visit docs.latch.bio to learn more.
+    """
+    try:
+        _get_params(wf_name, version)
+    except Exception as e:
+        raise e
```
[the remark this answers is not in the document]
Comment (kennyworkman) [reply latch/cli/main.py:202]: agree, leftover from debugging

File: latch/services/get_params.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+class Unsupported:
+    ...
+
+
+def _guess_python_type(literal: LiteralType, param_name: str):
+    """Transform flyte type literal to native python type."""
+
+    if literal.simple is not None:
+        simple_table = {
```
Comment (AidanAbd) [inline latch/services/get_params.py:164]: put this outside of function

**Review by AidanAbd — APPROVED** [review]
lgtm other than small comment

File: latch/cli/main.py [hunk 3]
```diff
@@ -181,6 +183,32 @@ def execute(params_file: Path, version: Union[str, None] = None):
...
+)
+def get_params(wf_name: Union[str, None], version: Union[str, None] = None):
+    """Generate a python parameter map for a workflow.
+
+    Visit docs.latch.bio to learn more.
+    """
+    try:
+        _get_params(wf_name, version)
+    except Exception as e:
+        click.secho(
```
Comment (maximsmol) [inline latch/cli/main.py:201]: print or save stacktrace to file

File: latch/services/__init__.py [hunk 4]
```diff
@@ -8,6 +8,7 @@
 from latch.services.cp import cp
 from latch.services.execute import execute
 from latch.services.get import get_wf
+from latch.services.get_params import get_params
```
Comment (maximsmol) [inline latch/services/__init__.py:11]: unused?

File: latch/services/get_params.py [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+_primitive_table = {
+    type(None): None,
+    int: 0,
+    float: 0.0,
+    str: "foo",
+    bool: False,
+}
+
+
+def get_params(wf_name: Union[None, str], wf_version: Union[None, str] = None):
```
Comment (maximsmol) [inline latch/services/get_params.py:47]:
```suggestion
def get_params(wf_name: Optional[str], wf_version: Optional[str] = None):
```

File: latch/services/get_params.py [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+        LatchFile: "from latch.types import LatchFile",
+        LatchDir: "from latch.types import LatchDir",
+        enum.Enum: "from enum import Enum",
+    }
+
+    import_types = []
+    enum_literals = []
+    param_map_str = ""
+    param_map_str += "\nparams = {"
+    param_map_str += f'\n    "_name": "{wf_name}", # Dont edit this value.'
```
Comment (maximsmol) [inline latch/services/get_params.py:100]:
```suggestion
    param_map_str += f'\n    "_name": "{wf_name}", # Don\'t edit this value.'
```

File: latch/services/get_params.py [hunk 7]
```diff
@@ -0,0 +1,353 @@
...
+        python_type, python_val, default = value
+
+        # Check for imports.
+
+        def _check_and_import(python_type: typing.T):
+            if python_type in import_statements and python_type not in import_types:
+                import_types.append(python_type)
+
+        def _handle_enum(python_type: typing.T):
+            if type(python_type) is enum.EnumMeta:
```
Comment (maximsmol) [inline latch/services/get_params.py:111]:
```suggestion
            if issubclass(python_type, Enum):
```

File: latch/services/get_params.py [hunk 8]
```diff
@@ -0,0 +1,353 @@
...
+                if enum.Enum not in import_types:
+                    import_types.append(enum.Enum)
+
+                variants = python_type._variants
+                name = python_type._name
+
+                _enum_literal = f"class {name}(Enum):"
+                for variant in variants:
+                    _enum_literal += f"\n    {variant} = '{variant}'"
```
Comment (maximsmol) [inline latch/services/get_params.py:120]: black uses double quotes for strings

File: latch/services/get_params.py [hunk 9]
```diff
@@ -0,0 +1,353 @@
...
+                enum_literals.append(_enum_literal)
+
+        # Parse collection, sum types for potential imports and dependent
+        # objects, eg. enum class construction.
+        if hasattr(python_type, "__origin__"):
```
Comment (maximsmol) [inline latch/services/get_params.py:125]: compare `get_origin` with `None` instead

File: latch/services/get_params.py [hunk 10]
```diff
@@ -0,0 +1,353 @@
...
+
+
+def _get_code_literal(python_val: any, python_type: typing.T):
+    """Construct value that is executable python when templated into a code
+    block."""
+
+    if python_type is str or (type(python_val) is str and str in get_args(python_type)):
+        return f'"{python_val}"', python_type
+
+    if type(python_type) is enum.EnumMeta:
```
Comment (maximsmol) [inline latch/services/get_params.py:169]:
```suggestion
    if issubclass(python_type, Enum):
```

File: latch/services/get_params.py [hunk 11]
```diff
@@ -0,0 +1,353 @@
...
+        name = python_type._name
+        return python_val, f"<enum '{name}'>"
+
+    if hasattr(python_type, "__origin__") and get_origin(python_type) is typing.Union:
```
Comment (maximsmol) [inline latch/services/get_params.py:173]:
```suggestion
    if get_origin(python_type) is typing.Union:
```

File: latch/services/get_params.py [hunk 12]
```diff
@@ -0,0 +1,353 @@
...
+        summands = get_args(python_type)
+        type_repr = "typing.Union["
+        for i, summand in enumerate(summands):
```
Comment (maximsmol) [inline latch/services/get_params.py:176]: use `", ".join([...])` instead

File: latch/services/get_params.py [hunk 13]
```diff
@@ -0,0 +1,353 @@
...
+            if i < len(summands) - 1:
+                delimiter = ", "
+            else:
+                delimiter = ""
+            type_repr += f"{_get_code_literal(python_val, summand)[1]}{delimiter}"
+        type_repr += "]"
+        return python_val, type_repr
+
+    if hasattr(python_type, "__origin__") and get_origin(python_type) is list:
```
Comment (maximsmol) [inline latch/services/get_params.py:185]:
```suggestion
    if get_origin(python_type) is list:
```

File: latch/services/get_params.py [hunk 14]
```diff
@@ -0,0 +1,353 @@
...
+        if python_val is None:
+            _, type_repr = _get_code_literal(None, get_args(python_type)[0])
+            return None, f"typing.List[{type_repr}]"
+        else:
+            collection_literal = "["
+            if len(python_val) > 0:
+                for i, item in enumerate(python_val):
```
Comment (maximsmol) [inline latch/services/get_params.py:192]: use `",".join([...])` instead

File: latch/services/get_params.py [hunk 15]
```diff
@@ -0,0 +1,353 @@
...
+
+    if literal.scalar is not None:
+        if literal.scalar.none_type is not None:
+            return None
+
+        if literal.scalar.primitive is not None:
+            primitive = literal.scalar.primitive
+
+            if primitive.string_value is not None:
+                if type(python_type) is enum.EnumMeta:
```
Comment (maximsmol) [inline latch/services/get_params.py:226]: issubclass

File: latch/services/get_params.py [hunk 16]
```diff
@@ -0,0 +1,353 @@
...
+
+        return typing.Union[tuple(summand_types)]
+
+    if literal.enum_type is not None:
+
+        # We can hold the variants a proxy class that is also type 'Enum', s.t.
+        # we can parse the variants and define the object in the param map
+        # code.
+
+        class _VariantCarrier(enum.Enum):
```
Comment (maximsmol) [inline latch/services/get_params.py:305]: kinda weird name

File: latch/services/get_params.py [hunk 17]
```diff
@@ -0,0 +1,353 @@
...
+        return []
+
+    file_like_table = {
+        LatchDir: LatchDir("latch:///foobar"),
+        LatchFile: LatchFile("latch:///foobar"),
+    }
+    if t in file_like_table:
+        return file_like_table[t]
+
+    if type(t) is enum.EnumMeta:
```
Comment (maximsmol) [inline latch/services/get_params.py:334]: issubclass

File: latch/services/get_params.py [hunk 18]
```diff
@@ -0,0 +1,353 @@
...
+        return f"{t._name}.{t._variants[0]}"
+
+    if not hasattr(t, "__origin__"):
```
Comment (maximsmol) [inline latch/services/get_params.py:337]:
```suggestion
    if get_origin(t) is None:
```

**Review by maximsmol — CHANGES_REQUESTED** [review]

**Pull request merged**